public/app/plugins/datasource/elasticsearch/language_provider.ts

```diff
@@ -104,22 +105,32 @@ export default class ElasticsearchLanguageProvider extends LanguageProvider {
     Object.assign(this, initialValues);
   }
 
+  /**
+   * The current implementation only supports switching from Prometheus/Loki queries.
+   * For them we transform the query to an ES Logs query since it's the behaviour most users expect.
+   * For every other datasource we just copy the refId and let the query editor initialize a default query.
+   * */
   importQueries(queries: DataQuery[], datasourceType: string): ElasticsearchQuery[] {
     if (datasourceType === 'prometheus' || datasourceType === 'loki') {
       return queries.map((query) => {
-        let prometheusQuery: PromQuery = query as PromQuery;
+        let prometheusQuery = query as PromQuery;
         const expr = getElasticsearchQuery(extractPrometheusLabels(prometheusQuery.expr));
         return {
           isLogsQuery: true,
```

Comment:
Do we still need `isLogsQuery: true` if we are adding ` type: 'logs'`?

Reply:
I'm not completely sure about it, i'm keeping it to avoid breaking anything, but for sure I'm gonna revisit it now with the BE refactoring. Thanks for pointing that out!